server/list.go

```diff
@@ -0,0 +1,342 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+const (
+	// MyListKey is the key used to store the order of the owned todos
+	MyListKey = ""
+	// InListKey is the key used to store the order of received todos
+	InListKey = "_in"
+	// OutListKey is the key used to store the order of sent todos
+	OutListKey = "_out"
+)
+
+// ListStore represents the KVStore operations for lists
+type ListStore interface {
```

Comment:
1/5 I am confused about this interface.

It'd appear to me that it should provide basic operations on `List` and `Task` keys, and leave the rest to the higher level API implementation.

Reply:
From my point of view, the only high level functions are:
- Pop and bump, which are here for atomicity
- GetIssueReference and GetIssueListAndReference which I feel they make things clearer if they are at this level, since they are not so much related with the business logic and more with the data shape.

Reply:
Ok, time will tell - we have the pattern of "index keys" in other plugins, hopefully a best practice of how to package it will materialize, we are a bit all over the place now.